src/core/linux/SDL_dbus.c

```diff
@@ -359,8 +377,68 @@ SDL_DBus_ScreensaverTickle(void)
 SDL_bool
 SDL_DBus_ScreensaverInhibit(SDL_bool inhibit)
 {
-    if ( (inhibit && (screensaver_cookie != 0)) || (!inhibit && (screensaver_cookie == 0)) ) {
+    const char *default_inhibit_reason = "Playing a game";
+
+    if ( (inhibit && (screensaver_cookie != 0 || inhibit_handle != NULL))
+        || (!inhibit && (screensaver_cookie == 0 && inhibit_handle == NULL)) ) {
         return SDL_TRUE;
+    }
+
+    /* Check if we are running inside a Flatpak or Snap container.
+     * For Snap, we check multiple variables because they might be set for unrelated
+     * reasons. This is the same thing WebKitGTK does. */
+    if (access("/.flatpak-info", F_OK) == 0
+        || (SDL_getenv("SNAP") != NULL && SDL_getenv("SNAP_NAME") != NULL && SDL_getenv("SNAP_REVISION") != NULL)) {
+
+        const char *node = "org.freedesktop.portal.Desktop";
```

Comment:
```suggestion
        const char *bus_name = "org.freedesktop.portal.Desktop";
```

is the standard D-Bus jargon for this